src/ride/transport/chairlift.c

```diff
@@ -13,3 +13,796 @@
  * A full copy of the GNU General Public License can be found in licence.txt
  *****************************************************************************/
 #pragma endregion
+
+#include "../../world/map.h"
+#include "../track_paint.h"
+#include "../track.h"
+#include "../../paint/paint.h"
+#include "../../interface/viewport.h"
+#include "../../paint/supports.h"
+#include "../ride.h"
+#include "../ride_data.h"
+
+enum
+{
+    SPR_METAL_FLOOR = 14567,
+    SPR_METAL_FENCE_NE = 14568,
+    SPR_METAL_FENCE_SE = 14569,
+    SPR_METAL_FENCE_SW = 14570,
+    SPR_METAL_FENCE_NW = 14571
+};
+
+enum
+{
+    SPR_CHAIRLIFT_CABLE_FLAT_SW_NE = 20500,
+    SPR_CHAIRLIFT_CABLE_FLAT_SE_NW,
+    SPR_20502,
+    SPR_20503,
+    SPR_20504,
+    SPR_20505,
+    SPR_CHAIRLIFT_STATION_COLUMN_NE_SW,
+    SPR_CHAIRLIFT_STATION_COLUMN_SE_NW,
+    SPR_20508,
+    SPR_20509,
+    SPR_20510,
+    SPR_20511,
+    SPR_20512,
+    SPR_20513,
+    SPR_20514,
+    SPR_20515,
+    SPR_CHAIRLIFT_CABLE_UP_SW_NE,
+    SPR_CHAIRLIFT_CABLE_UP_NW_SE,
+    SPR_CHAIRLIFT_CABLE_UP_NE_SW,
+    SPR_CHAIRLIFT_CABLE_UP_SE_NW,
+    SPR_20520,
+    SPR_20521,
+    SPR_20522,
+    SPR_20523,
+    SPR_20524,
+    SPR_20525,
+    SPR_20526,
+    SPR_20527,
+    SPR_CHAIRLIFT_CORNER_NW_SW,
+    SPR_CHAIRLIFT_CORNER_NW_NE,
+    SPR_CHAIRLIFT_CORNER_SE_NE,
+    SPR_CHAIRLIFT_CORNER_SW_SE,
+    SPR_20532,
+    SPR_20533,
+    SPR_20534,
+    SPR_20535,
+    SPR_20536,
+    SPR_20537,
+    SPR_20538,
+    SPR_20539,
+    SPR_CHAIRLIFT_BULLWHEEL_FRAME_1,
+    SPR_CHAIRLIFT_BULLWHEEL_FRAME_2,
+    SPR_CHAIRLIFT_BULLWHEEL_FRAME_3,
+    SPR_CHAIRLIFT_BULLWHEEL_FRAME_4,
+    SPR_CHAIRLIFT_STATION_END_CAP_NE,
+    SPR_CHAIRLIFT_STATION_END_CAP_SE,
+    SPR_CHAIRLIFT_STATION_END_CAP_SW,
+    SPR_CHAIRLIFT_STATION_END_CAP_NW,
+} SPR_CHAIRLIFT;
+
+const uint32 chairlift_bullwheel_frames[] = {
+    SPR_CHAIRLIFT_BULLWHEEL_FRAME_1,
+    SPR_CHAIRLIFT_BULLWHEEL_FRAME_2,
+    SPR_CHAIRLIFT_BULLWHEEL_FRAME_3,
+    SPR_CHAIRLIFT_BULLWHEEL_FRAME_4
+};
+
+enum
+{
+    SEGMENT_B4 = (1 << 0),
+    SEGMENT_0 = SEGMENT_B4,
+    SEGMENT_B8 = (1 << 1),
+    SEGMENT_1 = SEGMENT_B8,
+    SEGMENT_BC = (1 << 2),
+    SEGMENT_2 = SEGMENT_BC,
+    SEGMENT_C0 = (1 << 3),
+    SEGMENT_3 = SEGMENT_C0,
+    SEGMENT_C4 = (1 << 4),
+    SEGMENT_4 = SEGMENT_C4,
+    SEGMENT_C8 = (1 << 5),
+    SEGMENT_5 = SEGMENT_C8,
+    SEGMENT_CC = (1 << 6),
+    SEGMENT_6 = SEGMENT_CC,
+    SEGMENT_D0 = (1 << 7),
+    SEGMENT_7 = SEGMENT_D0,
+    SEGMENT_D4 = (1 << 8),
+    SEGMENT_8 = SEGMENT_D4,
+};
+
+const int SEGMENTS_ALL = SEGMENT_B4 | SEGMENT_B8 | SEGMENT_BC | SEGMENT_C0 | SEGMENT_C4 | SEGMENT_C8 | SEGMENT_CC | SEGMENT_D0 | SEGMENT_D4;
+
+static void paint_util_set_segment_support_height(int flags, uint16 height, uint8 segment_flags)
```

Comment:
We'll also need a segment rotation method, for tiles with a smaller footprint. Probably easiest to put the edge segments in an `uint8` and `ror8` them by `2 * rotation`. Not needed for this ride though, so not implemented yet